galley/pkg/config/analysis/analyzers/virtualservice/destinationhosts.go

```diff
@@ -112,34 +107,69 @@ func (d *DestinationHostAnalyzer) checkDestinationHost(vsNamespace string, desti
 		}
 
 		// Skip over entries not visible to the current virtual service namespace
-		if scope != util.ExportToAllNamespaces && scope != vsNamespace {
+		if scope != util.ExportToAllNamespaces && scope != sourceNs {
 			continue
 		}
 
 		seHostWithoutWildcard := strings.TrimPrefix(seHost, util.Wildcard)
 		hostWithoutWildCard := strings.TrimPrefix(host, util.Wildcard)
 
 		if strings.HasSuffix(hostWithoutWildCard, seHostWithoutWildcard) {
-			return true
+			return s
 		}
 	}
 
-	return false
+	return nil
 }
 
-func initServiceEntryHostNames(ctx analysis.Context) map[util.ScopedFqdn]bool {
-	hosts := make(map[util.ScopedFqdn]bool)
-	ctx.ForEach(metadata.IstioNetworkingV1Alpha3Serviceentries, func(r *resource.Entry) bool {
+func initServiceEntryHostMap(ctx analysis.Context) map[util.ScopedFqdn]*v1alpha3.ServiceEntry {
```

Comment:
This confused me quite a bit, calling this `initServiceEntryHostMap` even though this is a consolidation of all Platform Services and ServiceEntries. Suggest changing the naming:

`initServiceHostMap`

Reply:
The ServiceEntry name is because both the CR ServiceEntries and the discovered platform services both end up getting resolved as ServiceEntry objects. 

There's another PR coming down the pipe that moves this to a util function and renames it, I've punted the discussion of what to name it there: https://github.com/istio/istio/pull/18364